sdv/single_table/base.py

```diff
@@ -42,3 +46,175 @@ def get_parameters(self):
     def get_metadata(self):
         """Return the ``SingleTableMetadata`` for this synthesizer."""
         return self.metadata
+
+    def _validate_metadata_matches_data(self, columns):
+        errors = []
+        metadata_columns = self.metadata._columns if self.metadata._columns else []
+        invalid_data_columns = set(columns).difference(metadata_columns)
+        if invalid_data_columns:
+            errors.append(
+                f'The columns {sorted(invalid_data_columns)} are not present in the metadata.')
+
+        invalid_metadata_columns = set(metadata_columns).difference(columns)
```

Comment:
I think this should be called `missing_columns` based on the error message. They aren't invalid